### PR TITLE
8253434: Shenandoah: Cleanup ShenandoahRootScanner

### PR DESCRIPTION
**Pull request by zhengyu123**
After moving CLDG mark to concurrent phase, ShenandoahRootScanner::strong_roots_do() and ShenandoahRootScanner::roots_do() are practical the same, let's trim one.

Test:
  hotspot_gc_shenandoah

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253434](https://bugs.openjdk.java.net/browse/JDK-8253434): Shenandoah: Cleanup ShenandoahRootScanner

### Reviewers
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - **Reviewer**) ⚠️ Review applies to 2b40913b8ab5323e9421b48f836d84046366f804

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/286/head:pull/286`
`$ git checkout pull/286`

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp [hunk 1]
```diff
@@ -189,36 +189,13 @@ ShenandoahRootScanner::~ShenandoahRootScanner() {
 }
 
 void ShenandoahRootScanner::roots_do(uint worker_id, OopClosure* oops) {
-  CLDToOopClosure clds_cl(oops, ClassLoaderData::_claim_strong);
+  CLDToOopClosure clds_cl(oops, ClassLoaderData::_claim_none);
```
Comment (rkennke) [inline src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp:192]: Are you sure that we don't have to claim CLDs? How would the threads avoid doubly-marking CLDs?

Comment (rkennke) [reply src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp:192]: Actually it looks like the CLD closure is not used at all, or is it?

**Review by rkennke — COMMENTED** [review]
It looks to me you can also trim out the CLD closures altogether.

File: src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp (hunk 1 above)
Comment (zhengyu123) [reply src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp]: Right, we don't need to mark on-stack CLDs, concurrent mark CLDs should be sufficient.

Reran tests, still good.

File: src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp [hunk 2]
```diff
@@ -189,36 +189,12 @@ ShenandoahRootScanner::~ShenandoahRootScanner() {
...
-}
-
-void ShenandoahRootScanner::strong_roots_do(uint worker_id, OopClosure* oops) {
-  CLDToOopClosure clds_cl(oops, ClassLoaderData::_claim_strong);
-  MarkingCodeBlobClosure blobs_cl(oops, !CodeBlobToOopClosure::FixRelocations);
-  strong_roots_do(worker_id, oops, &clds_cl, &blobs_cl);
+  roots_do(worker_id, oops, NULL, &blobs_cl);
 }
 
 void ShenandoahRootScanner::roots_do(uint worker_id, OopClosure* oops, CLDClosure* clds, CodeBlobClosure* code, ThreadClosure *tc) {
```
Comment (rkennke) [inline src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp:196]: You're still passing a (NULL) CLDClosure here, but it's not used. Or am I missing something? The method argument could be eliminated altogether.

Comment (zhengyu123) [reply src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.cpp]: It is unused, removed.

Thanks

**Review by rkennke — APPROVED** [review]
Looks good to me now! Thank you!

**Pull request closed without merging**